### PR TITLE
Add Visual Search API swagger

### PR DESCRIPTION
**Pull request by shakao**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

File: specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json [hunk 1]
```diff
@@ -0,0 +1,2411 @@
+{
+    "parameters": {
+        "X-BingApis-SDK": "true",
+        "Content-Type": "application/json",
+        "Ocp-Apim-Subscription-Key": "{API key}",
+        "image": "multipart-form-data",
+        "knowledgeRequest": {
+            "imageInfo": {
+                imageInsightsToken: "{Image Insights Token}",
```
Comment (sergey-shandar) [inline specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json]: That's not a valid JSON.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json (hunk 1 above)
Comment (shakao) [reply specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json]: updated

**Comment by swhite-msft** [comment]
@shakao I don't think you want to merge this to master. I think you want it in the release-build-cognitive-search release branch.

**Comment by sudhansu88** [comment]
@swhite-msft : We have checked in all Specs to master only. What branch is this " release-build-cognitive-search" ?

**Comment by sudhansu88** [comment]
All well ? Any Changes remaining ?

File: specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json [hunk 2]
```diff
@@ -0,0 +1,2408 @@
...
+                                            "insightsMetadata":{
+                                                "recipeSourcesCount":0,
+                                                "pagesIncludingCount":8,
+                                                "availableSizesCount":3
+                                            },
+                                            "imageId":"FE52CFD00B7A8761700345AC0806EC23B1B256AD",
+                                            "accentColor":"956236"
+                                        },
+                                        {
+                                            "webSearchUrl":"https:\/\/www.bing.com\/images\/search?view=detailv2&FORM=OIIRPO
```
Comment (sergey-shandar) [inline specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json]: `_type` is missing.

Comment (sergey-shandar) [inline specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json]: `tags[*]._type` is expected, according to the swagger.

**Review by sergey-shandar — CHANGES_REQUESTED** [review]

File: specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json (hunk 2 above)
Comment (shakao) [reply specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json]: updated

Comment (shakao) [reply specification/cognitiveservices/data-plane/VisualSearch/v1.0/examples/SuccessfulVisualSearchRequest.json]: updated

File: specification/cognitiveservices/data-plane/VisualSearch/v1.0/VisualSearch.json [hunk 3]
```diff
@@ -0,0 +1,1253 @@
...
+            "required": false,
+            "type": "string"
+          },
+          {
+            "name": "knowledgeRequest",
+            "x-ms-client-name": "KnowledgeRequest",
+            "in": "formData",
+            "description": "A JSON object containing information about the image. The image and imageInsightsToken fields are mutually exclusive – the body of the request must include only one of them.",
+            "required": false,
+            "type": "string"
```
[the remark this answers is not in the document]
Comment (shakao) [reply specification/cognitiveservices/data-plane/VisualSearch/v1.0/VisualSearch.json]: We don't have the ability to specify objects in the body of a POST request using multipart form-data with Swagger 2.0, so it's encoded as a string for now. I've updated the example to match.

**Comment by sudhansu88** [comment]
@sergey-shandar : We are waiting for your sign off. Please let us know if the spec is ready to be merged. We will update Release branch Folks to pick final signed off version of spec and merge in Release branch instead of master.

**Comment by sergey-shandar** [comment]
@sudhansu88 The examples are still broken https://travis-ci.org/Azure/azure-rest-api-specs/jobs/367436243#L770

**Comment by shakao** [comment]
@sergey-shandar the first failure is because it's expecting a type "file", and the example placeholder is a string, which i based off the existing [Custom Vision API example](https://github.com/Azure/azure-rest-api-specs/blob/master/specification/cognitiveservices/data-plane/CustomVision/Prediction/stable/v1.1/examples/SuccessfulPredictionWithImage.json). i don't think there's a way to include a file upload in the example json?

**Comment by sergey-shandar** [comment]
@shakao yes, it's a known bug. Please ignore the first one and have a look the response.

**Review by sergey-shandar — APPROVED** [review]

**Pull request merged**